File: Homework06/0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.cs
public class Solution {
    public int StrStr(string haystack, string needle) {
        if(needle.Length == 0)
        {
            return 0;
        }

        for (int i = 0; i <= haystack.Length - needle.Length; i++)
        {
            if(haystack.Substring(i, needle.Length) == needle)
            {
                return i;
            }
        }
        return -1;
    }
}

// Time Complexity: O(m*n)

File: Homework06/0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:

        if(len(needle) == 0):
            return 0
        
        for i in range(len(haystack) + 1 - len(needle) ):
            if(haystack[i: len(needle) + i] == needle):
                return i
        return -1

            # Time Complexity: O(m*n)
